### PR TITLE
[NLP] ArtifactItem with init=True to make it debuggable

### PR DESCRIPTION
**Pull request by janekl**
# What does this PR do ?

When debugging artifact items I struggled with `ArtifactItem` as it is defined with `init=False` which results in issues when printing an instance of this class.

The following minimal example illustrates the problem - this code
```
from dataclasses import dataclass


@dataclass(init=False)
class Point:
    x: int
    y: int

p = Point()
print(p)
```
results in **AttributeError: 'Point' object has no attribute 'x'**. So it doesn't make it easier to check what's going on.

**Collection**: [NLP]

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] Improvement

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to https://github.com/NVIDIA/NeMo/pull/7935

### CODE REVIEW

File: nemo/core/connectors/save_restore_connector.py [hunk 1]
```diff
@@ -359,8 +359,6 @@ def register_artifact(self, model, config_path: str, src: str, verify_src_exists
         """
         app_state = AppState()
 
-        artifact_item = model_utils.ArtifactItem()
```
[the remark this answers is not in the document]
Comment (janekl) [reply nemo/core/connectors/save_restore_connector.py]: Good suggestion, adopted it. Now changes boil down to removing `init=False` and setting dummy defaults

**Review by titu1994 — APPROVED** [review]
Thanks LGTM

**Comment by titu1994** [comment]
jenkins

**Comment by janekl** [comment]
> Thanks LGTM

Thanks! I've just rebased it, I'll merge once pipeline completes.

**Pull request merged**